pallets/stablewap/src/types.rs

```diff
@@ -0,0 +1,148 @@
+use sp_runtime::Permill;
+use sp_std::ops::Add;
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_core::RuntimeDebug;
+use sp_runtime::traits::{CheckedAdd, Zero};
+
+pub type Balance = u128;
```

Comment:
Do we need different Balance type here than the one we set in runtime?

Reply:
i prefer to have the type in the pallet, so it does not depend on anything from runtime. 

in this way, it is standalone pallet and you can just take the pallet and move it elsewhere. 

the math and everything is done for u128 only. so it makes sense to have it directly here.

Reply:
That's fine it is a question then if we should either change this in xyk and lbp (As these use primitives:Balance) or make it configurable everywhere from the runtime.

Reply:
i consider primitives as types used in runtime. 

so if a pallet has configurable Balance type - it uses Balance from primitives.

The difference compared to have Balance type in this pallet ( or other pallet ) - is because the implementation is done for that type, and not for types from other package.

so if this type changes in the primitives for some reason - the pallet's implementation done for other types would not likely be correct anymore. 

I changed the visiblity of this type - and it is no longer public, but only pub(crate).